x-pack/plugins/cloud_integrations/cloud_experiments/public/launch_darkly_client/launch_darkly_client.ts

```diff
@@ -22,46 +23,81 @@ export interface LaunchDarklyUserMetadata
 }
 
 export class LaunchDarklyClient {
-  private launchDarklyClient?: LDClient;
+  private initiated = false;
+  private canceled = false;
+  private launchDarklyClient = new BehaviorSubject<LDClient | null>(null);
+  private loadingClient$ = new BehaviorSubject<boolean>(true);
+  private _launchDarklyClient$ = this.launchDarklyClient.asObservable();
 
   constructor(
     private readonly ldConfig: LaunchDarklyClientConfig,
     private readonly kibanaVersion: string
   ) {}
 
   public async updateUserMetadata(userMetadata: LaunchDarklyUserMetadata) {
+    if (this.canceled) return;
+
     const { userId, ...userMetadataWithoutUserId } = userMetadata;
     const launchDarklyUser: LDSingleKindContext = {
       ...userMetadataWithoutUserId,
       kind: 'user',
       key: userId,
     };
-    if (this.launchDarklyClient) {
-      await this.launchDarklyClient.identify(launchDarklyUser);
+
+    let launchDarklyClient: LDClient | null = null;
+    if (this.initiated) {
```

Comment:
Nit: should we name this `initialized`?

Reply:
Addressed in https://github.com/elastic/kibana/pull/181124/commits/d9eb81b2a6ae5983614ad9ae9cc30e9f820f2e38